zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -241,12 +241,12 @@ private String getSubDomain(String subdomain, String id) throws OrchestratorZone
         return subdomain;
     }
 
-    private void checkOrchestratorZoneExists(String name) throws ZoneAlreadyExistsException {
+    private void checkOrchestratorZoneExists(String name) throws OrchestratorZoneServiceException {
         IdentityZone identityZone = null;
         try{
             identityZone = zoneProvisioning.retrieveByName(name);
             if(identityZone != null){
-                throw new ZoneAlreadyExistsException("Orchestrator zone already exists for name:  " + name );
+                throw new OrchestratorZoneServiceException("Orchestrator zone already exists for name:  " + name );
```

Comment:
Here, you are throwing OrchestratorZoneServiceException but in catch block and controller error handler you re trying to handle ZoneAlreadyExistsException and same mentioned in commit message. Could you please recheck or explain how conflict handing is done?

Reply:
As discussed, for basic uniqueness check for zone name, there is a method checkOrchestratorZoneExists in side service class had earlier throw new ZoneAlreadyExistsException("Orchestrator zone already exists for name:  " + name ); which had been handled in controller as badRequest 400 status code, so while making changes on story "Create orchestrator zone API return conflict status code 409 on duplicate subdomain" , Hence, i have changed  "throw new ZoneAlreadyExistsException("Orchestrator zone already exists for name:  " + name );" to throw new OrchestratorZoneServiceException("Orchestrator zone already exists for name:  " + name );
   into checkOrchestratorZoneExists in side service class, so that in controller handle exception for both status code can be handled separately and accordingly subdomain uniqueness will throw new ZoneAlreadyExistsException(errorMessage); as below with status code 409 conflict
{
    "message": "The \"subdomain\" name is already taken. Please use a different subdomain."
}

Reply:
Done. Thanks!